core/services/gateway/handler.go

```diff
@@ -0,0 +1,46 @@
+package gateway
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/smartcontractkit/chainlink/v2/core/services/job"
+)
+
+type UserCallback interface {
+	SendResponse(msg *Message, errCode ErrorCode, errMsg string) error
```

Comment:
We probably need to require thread-safety for SendResponse, define blocking/non-blocking requirements, context, etc. Or replace the callback concept with a channel to avoid all of that.

Reply:
Yes, SendResponse is supposed to be called at most once and is non-blocking, I didn't make it clear. You're right that a channel might be a better idea. I thought that getting an error back could be valuable but for user HTTP calls there won't be any re-tries anyway (a handler can choose to support repeated requests). So let me convert that to a channel for now.